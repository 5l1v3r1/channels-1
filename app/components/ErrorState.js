// @flow
import React from 'react';
import styles from './ErrorState.scss';

const ErrorState = ({
  message,
  retry
}: {
  message: string,
  retry?: () => Promise<any>
}) => (
  <p className={styles.message}>
    {message}
    {retry &&
      <span className={styles.retry} onClick={retry}>
        {' '}
        Try again.
      </span>
    }
  </p>
);

export default ErrorState;
