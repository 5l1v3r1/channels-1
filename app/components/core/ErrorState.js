// @flow
import React from 'react';
import styles from './ErrorState.scss';

const ErrorState = ({
  message,
  retry,
  className
}: {
  message: string,
  retry?: () => Promise<any>,
  className?: string
}) => (
  <div className={[styles.errorState, className].join(' ')}>
    <p className={styles.message}>
      {message}
      {retry &&
        <span className={styles.retry} onClick={retry}>
          {' '}
          Try again.
        </span>
      }
    </p>
  </div>
);

export default ErrorState;
