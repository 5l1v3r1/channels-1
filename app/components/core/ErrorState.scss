@import '../../app.global.scss';

.errorState {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}

.message {
  margin: 0;
  padding: 1rem;
}

.retry {
  color: $gray;
  font-weight: 500;
}
