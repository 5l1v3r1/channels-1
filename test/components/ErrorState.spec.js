import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ErrorState from '../../app/components/core/ErrorState';

function setup(optionalProps = {}) {
  const props = {
    message: 'Error requesting channels.',
    color: 'black',
    ...optionalProps
  };

  const enzymeWrapper = shallow(<ErrorState {...props} />);

  return {
    props,
    enzymeWrapper
  };
}

describe('ErrorState', () => {
  describe('without retry', () => {
    it('should render self', () => {
      const { enzymeWrapper } = setup();
      const tree = toJson(enzymeWrapper);

      expect(tree).toMatchSnapshot();
    });
  });

  describe('with retry', () => {

    const optionalProps = {
      retry: jest.fn()
    };

    it('should render self', () => {
      const { enzymeWrapper } = setup(optionalProps);
      const tree = toJson(enzymeWrapper);

      expect(tree).toMatchSnapshot();
    });

    it('should call retry', () => {
      const { enzymeWrapper } = setup(optionalProps);
      const retry = enzymeWrapper.find('.retry');

      retry.simulate('click');

      expect(optionalProps.retry).toBeCalled();
    });
  });
});
