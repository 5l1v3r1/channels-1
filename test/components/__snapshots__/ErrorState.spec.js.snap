// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorState with retry should render self 1`] = `
<div
  className="errorState "
>
  <p
    className="message"
  >
    Error requesting channels.
    <span
      className="retry"
      onClick={[Function]}
    >
       
      Try again.
    </span>
  </p>
</div>
`;

exports[`ErrorState without retry should render self 1`] = `
<div
  className="errorState "
>
  <p
    className="message"
  >
    Error requesting channels.
  </p>
</div>
`;
